SPEC.md

```diff
@@ -182,6 +183,33 @@ Contents give more information about the issue's check, including a description
 
 Some engines require the ability to refer to other source locations. For this reason, the Issue type has an optional `other_locations` field, which is an array of other `Location` items that this issue needs to refer to.
 
+## Versioning
+
+This specification is versioned. The current version is [in the repository](https://github.com/codeclimate/spec/blob/master/VERSION). Engines declare the version of the specification they are compatible with in the manifest file, described below.
+
+## Manifest File
+
+All engines must include a `manifest.json` file at `/manifest.json`. This file, known as the manifest, includes information that is necessary for the analysis runtime. Here is an example of a manifest:
+
+```
+{
+  "name": "govet",
+  "maintainer": {
+    "name": "Michael R. Bernstein",
+    "email": "mrb@codeclimate.com",
+  },
+  "spec_version": "0.0.1",
+}
+```
+
+The fields in the manifest include:
+
+* `name` (`String`) - the name of the package.
+* `maintainer` (`Object`)
+  * `name` (`String`) - the name of the maintainer
+  * `email` (`String`) - the email address of the maintainer
+* `spec_version` (`String`) - the most recent version of the specification which this engine supports
```

Comment:
The answer to the first "What happens?" is:

* We bump the spec version to 1.0 (not backwards compatible). The Code Climate CLI, during that time, supports both the 1.x and 0.x SPEC. When it runs an engine, it looks at the declared `spec_version`, and if it's `0.x` it runs it with the old engine harness (probably emitting a warning)

The answer to the second "What happens?" is that we remove the `0.x` engine harness from the CLI, and it changes to refuse to run any pre-`1.x`-designated engines